### PR TITLE
Make Redis namespace fully configurable

### PR DESCRIPTION
**Pull request by driverpt**
New Clean PR from #598

### CODE REVIEW

File: spring-session-data-redis/src/main/java/org/springframework/session/data/redis/RedisOperationsSessionRepository.java [hunk 1]
```diff
@@ -588,7 +588,7 @@ private void publishEvent(ApplicationEvent event) {
 	}
 
 	public void setRedisKeyNamespace(String namespace) {
-		this.keyPrefix = DEFAULT_SPRING_SESSION_REDIS_PREFIX + namespace + ":";
+		this.keyPrefix = namespace + ":";
```
Comment (vpavic) [inline spring-session-data-redis/src/main/java/org/springframework/session/data/redis/RedisOperationsSessionRepository.java]: We need to validate that provided namespace is not `null` or empty.

File: spring-session-data-redis/src/test/java/org/springframework/session/data/redis/RedisOperationsSessionRepositoryTests.java [hunk 2]
```diff
@@ -709,9 +710,55 @@ public void setRedisFlushModeNull() {
 		this.redisRepository.setRedisFlushMode(null);
 	}
 
-	private String getKey(String id) {
-		return "spring:session:sessions:" + id;
-	}
+
+    @Test
+    public void testDefaultRedisNamespace() {
```
Comment (vpavic) [inline spring-session-data-redis/src/test/java/org/springframework/session/data/redis/RedisOperationsSessionRepositoryTests.java]: No need to add test for default namespace, as other tests validate this already. OTOH we need to validate that namespace cannot be set to `null` or empty string as per previous comment.

File: spring-session-data-redis/src/main/java/org/springframework/session/data/redis/RedisOperationsSessionRepository.java (hunk 1 above)
Comment (driverpt) [reply spring-session-data-redis/src/main/java/org/springframework/session/data/redis/RedisOperationsSessionRepository.java]: How do you suggest to do that? Throw an Exception?

Comment (vpavic) [reply spring-session-data-redis/src/main/java/org/springframework/session/data/redis/RedisOperationsSessionRepository.java]: Using `org.springframework.util.Assert#hasText` - I see you addressed that, thanks.

**Comment by driverpt** [comment]
test this please

**Comment by driverpt** [comment]
@vpavic , i've updated the code to include the tests.

How do i rebuild without adding commits on Travis ?

**Comment by vpavic** [comment]
Thanks for the update @driverpt - you can trigger the new Travis CI build by amending your commit without updating its contents, and force pushing the changes to the remote:

```
$ git commit --amend --no-edit
$ git push --force
```

**Comment by driverpt** [comment]
Is the master branch breaking ?

**Comment by vpavic** [comment]
@driverpt Both `master` branch and your PR build are stable - thanks for the update.

**Comment by vpavic** [comment]
Thanks for the contribution @driverpt. This is merged to `master` in 5f23a41 and polished in 36d349f.

**Comment by driverpt** [comment]
👍

**Pull request closed without merging**